src/components/Signup.jsx

```diff
@@ -77,52 +34,86 @@ const tailFormItemLayout = {
   },
 };
 const Signup = () => {
-  const [form] = Form.useForm();
-  const onFinish = (values) => {
-    console.log('Received values of form: ', values);
+  const [signupFormData, setSignupFormData] = useState({
+    name: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const handleNameChange = (e) => {
+    setSignupFormData({ ...signupFormData, name: e.target.value });
   };
-  const prefixSelector = (
-    <Form.Item name="prefix" noStyle>
-      <Select
-        style={{
-          width: 70,
-        }}
-      >
-        <Option value="86">+86</Option>
-        <Option value="87">+87</Option>
-      </Select>
-    </Form.Item>
-  );
-  const suffixSelector = (
-    <Form.Item name="suffix" noStyle>
-      <Select
-        style={{
-          width: 70,
-        }}
-      >
-        <Option value="USD">$</Option>
-        <Option value="CNY">¥</Option>
-      </Select>
-    </Form.Item>
-  );
-  const [autoCompleteResult, setAutoCompleteResult] = useState([]);
-  const onWebsiteChange = (value) => {
-    if (!value) {
-      setAutoCompleteResult([]);
-    } else {
-      setAutoCompleteResult(['.com', '.org', '.net'].map((domain) => `${value}${domain}`));
+
+  const handleEmailChange = (e) => {
+    setSignupFormData({ ...signupFormData, email: e.target.value });
+  };
+
+  const handlePasswordChange = (e) => {
+    setSignupFormData({ ...signupFormData, password: e.target.value });
+  };
+
+  const handleConfirmPasswordChange = (e) => {
+    setSignupFormData({ ...signupFormData, confirmPassword: e.target.value });
+  };
+
+  const isFormDataValidCheck = () => {
+    const { name, email, password, confirmPassword } = signupFormData;
+    const isAllFieldsFilled = name && email && password && confirmPassword;
+    const isEmailValid = email.includes('@') && email.includes('.');
+    if (!isAllFieldsFilled) {
```

Comment:
I would replace this with a regex because checking for `@` and `.` is not enough for email validation (for example `@.` would be a valid mail